### ISSUE TITLE
[Explorer] Allow for any ERC20 token to be displayed, not just our dummy CollateralToken

### ISSUE THREAD

**Issue by pelsasser** [MARKETProtocol/Dapp#71]
MARKET Protocol allows for any ERC20 token to be used as the base token for collateral in our contracts.  Currently the `Explore Contracts` and the `Find Contracts` pages believe these must be `CollateralToken`s and not ERC20 tokens.   

This needs to be fixed to allow for an ERC20 token to be used and displayed correctly in these elements.  Additionally, we should provide some way for people to still be able to see the addresses of these tokens.

Tasks to complete this issues and receive bounty
### Explore Contracts

- [ ] Fix issues with ERC20 vs CollateralToken to allow functionality for any token implementing the ERC20 interface
- [ ] Create a field when the contracts are expanded that shows the base token's address (so user's can confirm, many tokens can have the same name and symbol)

### Find Contracts

- [ ] Fix issues with ERC20 vs CollateralToken to allow functionality for any token implementing the ERC20 interface
- [ ] Create a field that shows the base token's address (so user's can confirm, many tokens can have the same name and symbol)

**If you are already working on another open or outstanding bounty please do not start work on this one until the other one has been paid out.  We would like to encourage many participants and parallelize our workflow as much as possible.  Thanks for the interest in our project and we look forward to working with you**

**Comment by pelsasser** [comment]
MARKET Protocol allows for any ERC20 token to be used as the base token for collateral in our contracts.  Currently the `Explore Contracts` and the `Find Contracts` pages believe these must be `CollateralToken`s and not ERC20 tokens.   

This needs to be fixed to allow for an ERC20 token to be used and displayed correctly in these elements.  Additionally, we should provide some way for people to still be able to see the addresses of these tokens.

Tasks to complete this issues and receive bounty
### Explore Contracts

- [ ] Fix issues with ERC20 vs CollateralToken to allow functionality for any token implementing the ERC20 interface
- [ ] Create a field when the contracts are expanded that shows the base token's address (so user's can confirm, many tokens can have the same name and symbol)

### Find Contracts

- [ ] Fix issues with ERC20 vs CollateralToken to allow functionality for any token implementing the ERC20 interface
- [ ] Create a field that shows the base token's address (so user's can confirm, many tokens can have the same name and symbol)

**If you are already working on another open or outstanding bounty please do not start work on this one until the other one has been paid out.  We would like to encourage many participants and parallelize our workflow as much as possible.  Thanks for the interest in our project and we look forward to working with you**

**Comment by gitcoinbot** [comment]
__This issue now has a funding of 0.125 ETH (90.4 USD @ $723.18/ETH) attached to it.__

 * If you would like to work on this issue you can claim it [here](https://gitcoin.co/issue/MARKETProtocol/Dapp/71).
 * If you've completed this issue and want to claim the bounty you can do so [here](https://gitcoin.co/issue/MARKETProtocol/Dapp/71)
 * Questions? Get help on the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $5084.34 more Funded OSS Work Available at: https://gitcoin.co/explorer

**Comment by maektwain** [comment]
Would like to work on this issue.

1. Explorer token 

You want the address field?

2. Find token

You want the token address to be returned associated ?

**Comment by gitcoinbot** [comment]
__Work has been started on the 0.125 ETH (87.23 USD @ $697.88/ETH) funding by__: 
 1. [@maektwain](https://gitcoin.co/profile/maektwain) 

 __Please work together__ and coordinate delivery of the issue scope. Gitcoin doesn't know enough about everyones skillsets / free time to say who should work on what, but we trust that the community is smart and well-intentioned enough to work together.  As a general rule; if you start work first, youll be at the top of the above list ^^, and should have 'dibs' as long as you follow through. 

 On the above list? Please leave a comment to let the funder (@pelsasser) and the other parties involved what you're working, with respect to this issue and your plans to resolve it.  If you don't leave a comment, the funder may expire your submission at their discretion. 

 * Learn more [on the gitcoin issue page](https://gitcoin.co/issue/MARKETProtocol/Dapp/71)
 * Questions? Get help on the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $3957.31 more Funded OSS Work Available at: https://gitcoin.co/explorer

**Comment by maektwain** [comment]
Hey, I am able to deploy the application in the dev env.

I am able to run the app, by first starting the bridge as mentioned and then dAPP.

I am not able to use the dApp properly it is showing web3 not initialised? Maybe I am missing something. 

Please point me some steps if I am missing? While deploying ?

**Comment by eswarasai** [comment]
@maektwain -- Let's connect over [Gitcoin Slack](https://gitcoincommunity.slack.com). I've DM'ed you

**Comment by maektwain** [comment]
Sure

**Comment by gitcoinbot** [comment]
__Work has been started on the 0.125 ETH (86.36 USD @ $690.86/ETH) funding by__: 
 1. [@maektwain](https://gitcoin.co/profile/maektwain) 

 __Please work together__ and coordinate delivery of the issue scope. Gitcoin doesn't know enough about everyones skillsets / free time to say who should work on what, but we trust that the community is smart and well-intentioned enough to work together.  As a general rule; if you start work first, youll be at the top of the above list ^^, and should have 'dibs' as long as you follow through. 

 On the above list? Please leave a comment to let the funder (@pelsasser) and the other parties involved what you're working, with respect to this issue and your plans to resolve it.  If you don't leave a comment, the funder may expire your submission at their discretion. 

 * Learn more [on the gitcoin issue page](https://gitcoin.co/issue/MARKETProtocol/Dapp/71)
 * Questions? Get help on the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $4183.5 more Funded OSS Work Available at: https://gitcoin.co/explorer

**Comment by vs77bb** [comment]
@maektwain Excited to have you working through Gitcoin. You're in great hands with @eswarasai and @pelsasser!

**Comment by maektwain** [comment]
#73

**Comment by eswarasai** [comment]
@pelsasser -- PR #77 solves this issue. But we also need to validate in the Deploy Contract form, whether the given Token Address is a valid `CollateralToken`/`ERC20 Token` or not. I'm gonna create a separate issue for this

**Comment by pelsasser** [comment]
@maektwain - I think you need to submit work on gitcoin to allow me to pay this bounty out.

Thanks for you hard work and interest in our project!

**Comment by gitcoinbot** [comment]
__The funding of 0.125 ETH (72.98 USD @ $583.88/ETH) attached to this issue has been approved & issued.__  

 * Learn more at [on the gitcoin issue page](https://gitcoin.co/issue/MARKETProtocol/Dapp/71)
 * Questions? Get help on the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $3795.52 more Funded OSS Work Available at: https://gitcoin.co/explorer

**Comment by maektwain** [comment]
@pelsasser Thanks for your support! and @eswarasai for providing help.

**Comment by pelsasser** [comment]
https://etherscan.io/tx/0x29d8776781cea1e9568fc91800ea15eb9d4b645ec7f67c11f884d7925252288a

**Comment by weberV** [comment]
Issue Status: 1. Open 2. **Expired** 

<hr>

__This issue now has a funding of 0.125 ETH (86.51 USD @ $692.1/ETH) attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](http://localhost:8000/issue/MARKETProtocol/Dapp/71/122).
 * Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $1,512.27 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)
